src/py/mat3ra/made/tools/build/nanoribbon/configuration.py

```diff
@@ -0,0 +1,38 @@
+from typing import Literal
+
+from mat3ra.made.material import Material
+from pydantic import BaseModel
+from mat3ra.code.entity import InMemoryEntity
+
+
+class NanoribbonConfiguration(BaseModel, InMemoryEntity):
+    """
+    Configuration for building a nanoribbon.
+
+    Attributes:
+        material (Material): The material to build the nanoribbon from.
+        width (int): The width of the nanoribbon in number of unit cells.
+        length (int): The length of the nanoribbon in number of unit cells.
+        edge_type (Literal["armchair", "zigzag"]): The edge type of the nanoribbon.
+    """
+
+    material: Material
+    width: int  # in number of unit cells
+    length: int  # in number of unit cells
+    vacuum_width: int = 3  # in number of unit cells
+    vacuum_length: int = 0  # in number of unit cells
+    edge_type: Literal["armchair", "zigzag"]
+
+    class Config:
+        arbitrary_types_allowed = True
```

Comment:
Move to BaseModel

Reply:
BaseModel is Pydantic's class, we can't adjust it. We don't have BaseConfiguration either, because there's no common properties. We can create a BaseConfig that'll house this Pydantic setup.